src/wrappers/AppInitializer.tsx

```diff
@@ -36,29 +36,36 @@ export function AppInitializer({
   const dispatch = useDispatch();
 
   async function initializeNetwork() {
+    const fetchConfigFromServer = !customNetworkConfig?.skipFetchFromServer;
+    const customNetworkApiAddress = customNetworkConfig?.apiAddress;
     const fallbackConfig = fallbackNetworkConfigurations[environment];
-    if (fallbackConfig != null) {
-      const newNetworkConfig = { ...fallbackConfig, ...customNetworkConfig };
-      dispatch(initializeNetworkConfig(newNetworkConfig));
-      initializeProviders(newNetworkConfig);
-    }
 
-    const customNetworkApiAddress = customNetworkConfig?.apiAddress;
-    const fallbackApiAddress = fallbackConfig?.apiAddress;
+    const localConfig = {
+      ...fallbackConfig,
+      ...customNetworkConfig
+    };
+
+    if (fetchConfigFromServer) {
+      const fallbackApiAddress = fallbackConfig?.apiAddress;
 
-    if (!customNetworkConfig?.skipFetchFromServer) {
       const serverConfig = await getServerConfiguration(
         customNetworkApiAddress || fallbackApiAddress
       );
 
       if (serverConfig != null) {
-        const configFromServer = { ...fallbackConfig, ...customNetworkConfig };
-        dispatch(
-          initializeNetworkConfig({ ...serverConfig, ...customNetworkConfig })
-        );
-        initializeProviders(configFromServer);
+        const apiConfig = {
+          ...(fallbackConfig ? { ...fallbackConfig } : {}),
```

Comment:
si poti sa scapi de asta daca faci handle for null la initializare

Reply:
✅